Gems/EMotionFX/Code/EMotionFX/Source/DualQuatSkinDeformer.cpp

```diff
@@ -79,37 +77,19 @@ namespace EMotionFX
     {
         const Actor* actor = actorInstance->GetActor();
         const Pose* pose = actorInstance->GetTransformData()->GetCurrentPose();
-        const uint32 numVertices = m_mesh->GetNumVertices();
 
-        // pre-calculate the skinning matrices
+        // Calculate the skinning matrices based on the current pose.
         for (BoneInfo& boneInfo : m_bones)
         {
             const size_t nodeIndex = boneInfo.m_nodeNr;
             const Transform skinTransform = actor->GetInverseBindPoseTransform(nodeIndex) * pose->GetModelSpaceTransform(nodeIndex);
             boneInfo.m_dualQuat.FromRotationTranslation(skinTransform.m_rotation, skinTransform.m_position);
         }
 
-        AZ::JobCompletion jobCompletion;
-
-        // Split up the skinned vertices into batches.
-        const AZ::u32 numBatches = aznumeric_caster(ceilf(aznumeric_cast<float>(numVertices) / aznumeric_cast<float>(s_numVerticesPerBatch)));
-        for (AZ::u32 batchIndex = 0; batchIndex < numBatches; ++batchIndex)
-        {
-            const AZ::u32 startVertex = batchIndex * s_numVerticesPerBatch;
-            const AZ::u32 endVertex = AZStd::min(startVertex + s_numVerticesPerBatch, numVertices);
-
-            // Create a job for every batch and skin them simultaneously.
-            AZ::JobContext* jobContext = nullptr;
-            AZ::Job* job = AZ::CreateJobFunction([this, startVertex, endVertex]()
-                {
-                    SkinRange(m_mesh, startVertex, endVertex, m_bones);
-                }, /*isAutoDelete=*/true, jobContext);
-
-            job->SetDependent(&jobCompletion);
-            job->Start();
-        }
-
-        jobCompletion.StartAndWaitForCompletion();
+        // Skin the vertices by executing the task graph.
```

Comment:
Looks great! One request that will unfortunately uglify the code somewhat is to reintroduce the old code and branch between job/task systems using the `cl_activateTaskGraph` cvar. This cvar is being used elsewhere as systems migrate so we can do A/B comparisons while the new system hardens and changes are introduced to the old system.

Reply:
Changed. We now have both versions in and the decision is done based on the cvar.